library/lua/gui/widgets/text_area/wrapped_text.lua

```diff
@@ -0,0 +1,70 @@
+-- This class caches lines of text wrapped to a specified width for performance
+-- and readability. It can convert a given text index to (x, y) coordinates in
+-- the wrapped text and vice versa.
+
+-- Usage:
+-- This class should only be used in the following scenarios.
+--  1. When text or text features need to be rendered
+--     (wrapped {x, y} coordinates are required).
+--  2. When mouse input needs to be converted to the original text position.
+
+-- Using this class in other scenarios may lead to issues with the component's
+-- behavior when the text is wrapped.
+WrappedText = defclass(WrappedText)
+
+WrappedText.ATTRS{
+    text = '',
+    wrap_width = DEFAULT_NIL,
+}
+
+function WrappedText:init()
+    self:update(self.text, self.wrap_width)
+end
+
+function WrappedText:update(text, wrap_width)
+    self.lines = text:wrap(
+        wrap_width,
```

Comment:
if `wrap_width` is `nil`, then the text will be wrapped at 72 characters. Is this appropriate? Should `wrap_width` instead default to `math.huge`?